dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/tracker/PotentialDuplicatesActions.java

```diff
@@ -44,22 +44,38 @@ public PotentialDuplicatesActions()
         super( "/potentialDuplicates" );
     }
 
-    public ApiResponse createPotentialDuplicate( String teiA, String teiB, String status )
+    public String createAndValidatePotentialDuplicate( String teiA, String teiB, String status )
     {
         JsonObject object = new JsonObjectBuilder()
             .addProperty( "original", teiA )
             .addProperty( "duplicate", teiB )
-            .addProperty( "status", status )
+            .addProperty( "status", "OPEN" )
             .build();
 
-        return this.post( object );
+        String uid = this.post( object ).validateStatus( 200 ).extractUid();
+
+        if ( status.equals( "MERGED" ) )
+        {
+            this.autoMergePotentialDuplicate( uid ).validateStatus( 200 );
+        }
+
+        if ( status.equals( "INVALID" ) )
+        {
+            this.update( uid + "?status=INVALID", new JsonObjectBuilder().build() ).validateStatus( 200 );
+        }
+
+        return uid;
     }
 
-    public String createAndValidatePotentialDuplicate( String teiA, String teiB, String status )
+    public ApiResponse createPotentialDuplicate( String teiA, String teiB, String status )
```

Comment:
I think the naming here is a bit misleading. Since the contract changed, only OPEN status will **create** a duplicate. I would change the name to "postPotentialDuplicate", or add a javadoc explaining that.